persistence/aggregate.go

```diff
@@ -62,3 +87,37 @@ func (op SaveAggregateMetaData) AcceptVisitor(ctx context.Context, v OperationVi
 func (op SaveAggregateMetaData) entityKey() entityKey {
 	return entityKey{"handler", op.MetaData.HandlerKey, op.MetaData.InstanceID}
 }
+
+// SaveAggregateSnapshot is an Operation that creates or updates snapshots of
+// an aggregate instance.
+type SaveAggregateSnapshot struct {
+	// Snapshot is the snapshot to persist.
+	Snapshot AggregateSnapshot
+}
+
+// AcceptVisitor calls v.VisitSaveAggregateSnapshot().
+func (op SaveAggregateSnapshot) AcceptVisitor(ctx context.Context, v OperationVisitor) error {
+	return v.VisitSaveAggregateSnapshot(ctx, op)
+}
+
+func (op SaveAggregateSnapshot) entityKey() entityKey {
+	return entityKey{"handler", op.Snapshot.HandlerKey, op.Snapshot.InstanceID}
```

Comment:
```suggestion
	return entityKey{"snapshot", op.Snapshot.HandlerKey, op.Snapshot.InstanceID}
```

These operations will need "entity keys" distinct from `SaveAggregateMetaData`, otherwise they'll conflict and cause a panic when performed in the same `UnitOfWork`.

I suspect this simply isn't being caught because there's no end-to-end test of handler behaviour that uses real persistence providers.